app/Http/Controllers/TestResultController.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\User;
+use App\Test;
+use App\Result;
+use Illuminate\Http\Request;
+use App\Http\Resources\TestResult;
+use App\Http\Resources\TesteeTestResult;
+use App\Http\Resources\Result as ResultDetails;
+
+class TestResultController extends Controller
+{
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function getTestResultById($id)
+    {
+        $testResults = Test::findOrFail($id)->with('results')->get();
+
+        return TestResult::collection($testResults);
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function getTesteeResult($id)
+    {
+        $testResults = User::where('type', User::TYPE_TESTEE)->with('results')->whereId($id)->get();
+
+        return TesteeTestResult::collection($testResults);
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function getGroupResult($id)
```

Comment:
My opinion is the same for this method as for the previous one. Move this to a separate controller and change the way you're getting the data from DB: use directly "results" table.